### PR TITLE
feat:[SSCA-3019]: Added schema for Artifact Signing and Artifact Verification step

### PR DESCRIPTION
**Pull request by humanshu2002**
This pull request includes significant updates to the CI and security pipeline configurations, as well as the addition of several new artifact signing and verification specifications. The changes primarily focus on enhancing the artifact signing and verification processes within the pipeline.

### Enhancements to CI and Security Pipeline Configurations:

* [`v0/pipeline/stages/ci/execution-wrapper-config.yaml`](diffhunk://#diff-ee2635334066a37f9e1d774450dd48051e7ba2667d4646e63ba5bb2cb1b72a85R82-R83): Added references to `artifact-signing-step-node.yaml` and `artifact-verification-step-node.yaml` to enhance the CI pipeline with artifact signing and verification steps.
* [`v0/pipeline/stages/security/execution-wrapper-config.yaml`](diffhunk://#diff-25961101f4ae1230fe0558c70c7cf9c0fc89daa604a801c1fdb997922c05978fR66-R67): Included references to `artifact-signing-step-node.yaml` and `artifact-verification-step-node.yaml` to improve the security pipeline with artifact signing and verification steps.

### Addition of Artifact Signing Specifications:

* [`v0/pipeline/steps/common/artifact-signing-acr-source-spec.yaml`](diffhunk://#diff-78cf14a9beaeed653e99fb294b393396a798606403789bbcd99d9e9983026244R1-R18): Introduced `ArtifactSigningAcrSourceSpec` for ACR source specifications.
* [`v0/pipeline/steps/common/artifact-signing-docker-source-spec.yaml`](diffhunk://#diff-cedd72716e8e724c046bad53409f51ca549f6014aa2abac4f51745fcba5a896eR1-R16): Added `ArtifactSigningDockerSourceSpec` for Docker source specifications.
* [`v0/pipeline/steps/common/artifact-signing-ecr-source-spec.yaml`](diffhunk://#diff-23556d4a6e3c1ae0dc51c88573c69990f705ccc67bc46ed394bfffb1ff89217eR1-R22): Created `ArtifactSigningEcrSourceSpec` for ECR source specifications.
* [`v0/pipeline/steps/common/artifact-signing-gar-source-spec.yaml`](diffhunk://#diff-153ed5bd48c2e99cfa27eaec9d3344c610e64d55786048cef608aa8c61555306R1-R22): Defined `ArtifactSigningGarSourceSpec` for GAR source specifications.
* [`v0/pipeline/steps/common/artifact-signing-gcr-source-spec.yaml`](diffhunk://#diff-3fd5b303d8f0d03d1ef419e1b00ff46e8d62ce177b289fc8d134524b39731b42R1-R22): Added `ArtifactSigningGcrSourceSpec` for GCR source specifications.
* [`v0/pipeline/steps/common/artifact-signing-har-source-spec.yaml`](diffhunk://#diff-13f49aa0629543c84c7fddc7d03bde99c511868a351076dbeca6f16e3b113ef4R1-R16): Introduced `ArtifactSigningHarSourceSpec` for HAR source specifications.

### Addition of Artifact Signing and Verification Steps:

* [`v0/pipeline/steps/common/artifact-signing-step-node.yaml`](diffhunk://#diff-eb5e4f2e82109a8cd20103ddbaa8a94b6c42651290f0cdfcf7d3811c624656d0R1-R55): Added `ArtifactSigningStepNode` to define the structure and properties for artifact signing steps.
* [`v0/pipeline/steps/common/artifact-verification-step-node.yaml`](diffhunk://#diff-26b70cdc6a11514cf44001dc08e201895372a8a603f645060c7c50c5b9183c1dR1-R55): Created `ArtifactVerificationStepNode` to outline the structure and properties for artifact verification steps.

### Addition of Cosign Signing Specifications:

* [`v0/pipeline/steps/common/cosign-signing.yaml`](diffhunk://#diff-e086692b563e65aca6d6a8a9713ca195749b27a5ced53e418129c923756b665aR1-R16): Introduced `CosignSigning` for defining the properties and requirements for Cosign signing.
* [`v0/pipeline/steps/common/cosign-secret-manager-signing.yaml`](diffhunk://#diff-92de77e89762caea31e4994c148a06d408a58a83adb80f87edf0f66c079e0755R1-R16): Added `CosignSecretManagerSigning` for specifying Cosign signing with Secret Manager.

### CODE REVIEW

File: v0/pipeline/steps/common/artifact-signing-docker-source-spec.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+title: ArtifactSigningDockerSourceSpec
+allOf:
+- $ref: artifact-signing-source-spec.yaml
+- type: object
+  required:
+    - image
+    - connector
```
Comment (reetika04) [inline v0/pipeline/steps/common/artifact-signing-docker-source-spec.yaml:7]: What if artifact type is harness?

File: v0/pipeline/steps/common/artifact-signing-source.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+type: object
+properties:
+  type:
+    type: string
+    enum:
+    - docker
+    - gcr
+    - ecr
+    - acr
+    - gar
```
Comment (reetika04) [inline v0/pipeline/steps/common/artifact-signing-source.yaml:11]: What about har?

File: v0/pipeline/steps/common/artifact-signing.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  - if:
+      properties:
+        type:
+          const: cosign
+    then:
+      properties:
+        spec:
+          type: object
+          oneOf:
+            - $ref: cosign-signing.yaml
```
Comment (reetika04) [inline v0/pipeline/steps/common/artifact-signing.yaml:23]: Should we keep this same everywhere since we are reusing pretty much everything.

Comment (humanshu2002) [reply v0/pipeline/steps/common/artifact-signing.yaml:23]: kept it same

**Review by reetika04 — APPROVED** [review]

**Review by harshjain12 — APPROVED** [review]

**Pull request merged**